test/utils/testUtils.ts

```diff
@@ -0,0 +1,819 @@
+import { faker } from "@faker-js/faker";
+import { currenciesByNetwork } from "@hypercerts-org/marketplace-sdk";
+import { Kysely, sql } from "kysely";
+import { DataType, newDb } from "pg-mem";
+import { getAddress } from "viem";
+import { expect } from "vitest";
+import { MarketplaceOrderSelect } from "../../src/services/database/entities/MarketplaceOrdersEntityService.js";
+import { CachingDatabase } from "../../src/types/kyselySupabaseCaching.js";
+import { DataDatabase } from "../../src/types/kyselySupabaseData.js";
+
+export type TestDatabase = CachingDatabase | DataDatabase;
+
+export async function createTestDataDatabase(
+  setupSchema?: (db: Kysely<DataDatabase>) => Promise<void>,
+) {
+  const mem = newDb();
+
+  // Create database instance
+  const db = mem.adapters.createKysely() as Kysely<DataDatabase>;
+
+  // NOTE: pg-mem does not support the generateUUID() function, so we need to register our own and for some reason it needs to be lowercase
+  mem.public.registerFunction({
+    name: "generateuuid",
+    returns: DataType.uuid,
+    implementation: () => faker.string.uuid(),
+  });
+
+  // NOTE: pg-mem does not support the array_append function, so we need to register our own
+  mem.public.registerFunction({
+    name: "array_append",
+    args: [
+      mem.public.getType(DataType.text).asArray(),
+      mem.public.getType(DataType.text),
+    ],
+    returns: mem.public.getType(DataType.text).asArray(),
+    implementation: (arr: string[], element: string) => [...arr, element],
+  });
+
+  mem.public.registerFunction({
+    name: "exists",
+    args: [mem.public.getType(DataType.uuid).asArray()],
+    returns: mem.public.getType(DataType.bool),
+    implementation: (arr: string[]) => arr.length > 0,
+  });
+
+  // Create marketplace_orders table
+  // TODO typings in DB are inconsisten do this will need to be updated when the DB is updated
```

Comment:
Typo. Also, when and why will the database be updated?

Reply:
Leave this to @Jipperism . I think the database tables need some cleanup and renaming, to make types clear and consisten between the two database. But, it would be some work so he should align with @holkexyz

Reply:
@bitbeckers can you remember which typings are inconsistent exactly? I'd like to put this on the backlog but not sure what to put in the issue rn.

Reply:
Created https://github.com/hypercerts-org/hypercerts-api/issues/282 to track this.